esp/esp/program/models/__init__.py

```diff
@@ -1326,16 +1341,21 @@ def getLastForProgram(user, program):
                                'student_info', 'teacher_info', 'guardian_info',
                                'educator_info')
                            .order_by('-last_ts','-id')[:1])
-        if len(regProfList) < 1:
-            regProf = RegistrationProfile.getLastProfile(user)
-            regProf.program = program
-            if regProf.id is not None:
-                regProf.id = None
-                if (datetime.now() - regProf.last_ts).days <= 5:
-                    regProf.save()
+        
+        if len(profile_list) < 1:
+            registration_profile = RegistrationProfile.getLastProfile(user)
+            registration_profile.program = program
+
+            if registration_profile.id is not None:
+                registration_profile.id = None
+                if (datetime.now() - registration_profile.last_ts).days <= 5:
```

Comment:
This may be where you intended to use PROFILE_MAX_AGE_DAYS (sorry about the extra comment).